content/en/blog/_posts/Introducing-Kueue.md

```diff
@@ -0,0 +1,199 @@
+most of Kueue’s operations are centered around ClusterQueue; the ResourceFlavor and LocalQueue APIs are mainly organizational wrappers.
+
+### Example Use Case
+Imagine the following setup for running batch workloads on a Kubernetes cluster on the cloud: 
+- We have two types of autoscaled node groups that differ on their provisioning policies: spot and on-demand. The nodes of each group are 
+  differentiated by the label `instance-type = {spot | ondemand}`. Moreover, since not all jobs can tolerate running on spot nodes, the nodes 
+  are tainted with spot=true:NoSchedule.
+- To strike a balance between cost and resource availability, we want jobs to use up to 1000 cores of on-demand nodes, then use up to 2000 cores 
+  of spot nodes.
+
+As batch admins, we define two ResourceFlavors that represent the two types of nodes
+
+```yaml
+apiVersion: kueue.x-k8s.io/v1alpha2
+kind: ResourceFlavor
+metadata:
+  name: ondemand
+  labels:
+    instance-type: ondemand 
+---
+apiVersion: kueue.x-k8s.io/v1alpha2
+kind: ResourceFlavor
+metadata:
+  name: spot
+  labels:
+    instance-type: spot
+taints:
+- effect: NoSchedule
+  key: spot
+  value: "true"
+```
+
+Then we define the quotas by creating a ClusterQueue as follows
+```yaml
+apiVersion: kueue.x-k8s.io/v1alpha2
+kind: ClusterQueue
+metadata:
+  name: research-pool
+spec:
+  namespaceSelector: {}
+  resources:
+  - name: "cpu"
+    flavors:
+    - name: ondemand
+      quota:
+        min: 1000
+    - name: spot
+      quota:
+        min: 2000
+```
+
+Note that the order of flavors in the ClusterQueue resources matters: Kueue will attempt to fit jobs in the available quotas according to 
+the order unless the job has an explicit affinity to specific flavors. 
+
+For each namespace, we define a LocalQueue that points to the ClusterQueue above.
+
+```yaml
+apiVersion: kueue.x-k8s.io/v1alpha2
+kind: LocalQueue
+metadata:
+  name: training
+  namespace: team-ml
+spec:
+  clusterQueue: research-pool
+```
+
+Admins create the above setup once. As batch users, we are able to find the queues we are allowed to submit to by listing the `LocalQueues` 
+in our namespace. The command is similar to the following: `kubectl get -n my-namespace localqueues`
+
+To submit a job, we simply set the `kueue.x-k8s.io/queue-name` annotation as follows:
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+metadata:
+  generateName: sample-job-
+  annotations:
+    kueue.x-k8s.io/queue-name: main
+spec:
+  parallelism: 3
+  completions: 3
+  template:
+    spec:
+      tolerations:
+      - key: spot
+        operator: "Exists"
+        effect: "NoSchedule"
+      containers:
+      - name: dummy-job
+        image: gcr.io/k8s-staging-perf-tests/sleep:latest
+        args: ["30s"]
+        resources:
+          requests:
+            cpu: 1
+      restartPolicy: Never
+```
+
+
+Kueue operates by forcing the job to be created in suspended mode. Once the job is at the head of the ClusterQueue, Kueue evaluates if it can start
```

Comment:
```suggestion
Kueue uses [admission control](/docs/reference/access-authn-authz/admission-controllers/) to ensure that
the Job is created in suspended mode. Once the job is at the head of the ClusterQueue, Kueue evaluates if
that work can start,
```

Reply:
We use a webhook and some reconciliation logic to enforce that, not really the builtin admission controllers

Reply:
Actually currently it is enforced by the job controller within Kueue not a webhook, we may add a webhook to do that on object creation in the future. I don't think this detail is important for the blog though.

Reply:
I didn't like the sentence “Kueue operates by forcing the job to be created in suspended mode” because it's not clear if the reader or the control plane is responsible for that enforcement.

Try to reword to make that detail clear.

Reply:
Is there a mechanism to stop the built-in job controller from acting on that Job?

Reply:
How about "Kueue ensures that the job is in suspended mode" without specifying how for now?

> Is there a mechanism to stop the built-in job controller from acting on that Job?

Currently the suspend flag is set to false when kueue's job controller first observes the job; so the built-in job controller could have started to create the pods if it observed the job first, but all that will be reversed when kueue gets the create event. This is not ideal, but we will add a webhook in the next release so that the job is created with suspend=true.

Reply:
OK, that works. I'd assumed the webhook implementation already existed, but no worries.

Even better would be to say something like:
“_Kueue intervenes to suspend the job as soon as it is created_” - which is more strictly true now and will stay true with the webhook implementation. That way, we're not risking a situation where someone seeing a bug assumes a feature that's not yet delivered. Readers might infer that Kueue _today_ configures a validating or mutating admission mechanism that doesn't actually exist.

Reply:
sounds good, done.